subsys/dfu/Kconfig

```diff
@@ -14,20 +14,34 @@ menuconfig IMG_MANAGER
 
 if IMG_MANAGER
 
-choice
-	prompt "Image manager"
-	default MCUBOOT_IMG_MANAGER
+config FLASH_IMG
+	bool "Flash Image API"
```

Comment:
need some better prompt here which could  help with understanding what this module really does: Flash image collection API?

Reply:
I agree. 

Ideally the flash_img.c file should be moved out of the dfu subsystem altogether, and renamed to something more generic. Like `flash_bufferd_write.c`, as this is actually what it does, it is not related to DFU at all.

I am willing to make this change if it is deemed valid by you guys.

Reply:
For me sounds OK.
I can suggest subsys/storage/flash_buffred_writer[or fbw] path.

Reply:
SGTM though the "buffered" typo should be fixed.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/23290

Here is a draft pull request with the generic module. I see that this is definitely something we need (think download to external flash, loading from external flash to internal flash, loading image from application core flash to network core flash on 53).

Reply:
> #23290
> 
> Here is a draft pull request with the generic module. I see that this is definitely something we need (think download to external flash, loading from external flash to internal flash, loading image from application core flash to network core flash on 53).

Thanks a lot! I've left some comments there. I think this is a great direction to move in.

I hope you're not too upset with me for widening the scope to include revisiting the flash_map.h API.